ONSdigital/eq-survey-runner
# Final session hardening

### Changes
1. Generating a User IK along with the additional SALT required to generate the encryption key
2. The user ik is now stored in the cookie along side the session id
3. The user id remains in the database
4. The data in the database is now encrypted with a combination the user id, user ik and a PEPPER
5. Session and Session mapping table are now cleared on log in to prevent session fixation attacks
### How to test
1. Check out this branch
2. Enable Server side storage
3. Complete a survey, check everything still works
4. Check save and resume still works.
### Who can test

Anyone apart from @warrenbailey



---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/onsdigital/eq-survey-runner/249)
